sklearn/metrics/pairwise.py

```diff
@@ -1120,18 +1121,27 @@ def cosine_distances(X, Y=None):
     array([[1.     , 1.     ],
            [0.42..., 0.18...]])
     """
+    xp, _ = get_namespace(X, Y)
+
     # 1.0 - cosine_similarity(X, Y) without copy
     S = cosine_similarity(X, Y)
     S *= -1
     S += 1
-    np.clip(S, 0, 2, out=S)
+    S = _clip(X, S, 0, 2, xp)
     if X is Y or Y is None:
         # Ensure that distances between vectors and themselves are set to 0.0.
         # This may not be the case due to floating point rounding errors.
-        np.fill_diagonal(S, 0.0)
+        _fill_diagonal_2d(S, 0.0, xp)
     return S
 
 
+def _fill_diagonal_2d(S, val, xp):
+    assert S.ndim == 2, "_fill_diagonal_2d supports 2D arrays only"
+    n, m = S.shape
```

Comment:
Since `n` does not seem to be used below
```suggestion
    _, m = S.shape
```

Reply:
Will change, thank you!